graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,211 @@
+        self, vertex_out_class_name, vertex_in_class_name, edge_class_name
+    ):
+        """Return number of edges of given class connecting two vertex types.
+
+        For cost estimation, sometimes we need to predict traversal costs between two vertex types
+        using an edge type that connects the vertices' abstract types. If this statistic isn't
+        provided, it can be roughly estimated using multiple calls of get_class_count().
+
+        Args:
+            vertex_out_class_name: str, vertex class name.
+            vertex_in_class_name: str, vertex class name.
+            edge_class_name: str, edge class name.
+
+        Returns:
+            - int, the count of edges if the statistic exists
+            - None otherwise
+        """
+        statistic_id = (vertex_in_class_name, vertex_out_class_name, edge_class_name)
+        if statistic_id not in self._edge_count_between_vertex_pair:
+            return None
+        return self._edge_count_between_vertex_pair[statistic_id]
+
+    def get_domain_count(self, vertex_name, field_name):
+        """Return the number of distinct values the vertex field has over all vertex instances.
+
+        This statistic helps estimate the result size of @filter directives used to restrict
+        values using equality operators like '=', '!=', and 'in_collection'.
+
+        Preconditions:
+        1. field_name must be a valid vertex field of vertex_name
+
+        Args:
+            vertex_name: str, name of a vertex.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - int, domain size of the vertex field's values if the statistic exists
+            - None otherwise
+        """
+        statistic_id = (vertex_name, field_name)
+        if statistic_id not in self._domain_count:
+            return None
+        return self._domain_count[statistic_id]
+
+    def get_histogram(self, vertex_name, field_name):
+        """Return a histogram for the given vertex field, providing statistics about range values.
+
+        This statistic helps estimate the result size of @filter directives used to restrict
+        numbers using inequality operators like '>', '<', and 'between'.
+
+        Preconditions:
+        1. field_name must be a valid vertex field of vertex_name
+
+        Args:
+            vertex_name: str, name of a vertex.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - list[tuple(int, int, int)], histogram as a list of tuples with 3 elements each if the
+              statistic exists. Each tuple represents a bin as (start, end, elementCount), where
+              [start, end) is the range of values in the bin, and elementCount is the bin element
+              count.
+            - None otherwise
+        """
+        if vertex_name not in self._histogram or field_name not in self._histogram[vertex_name]:
+            return None
+        return self._histogram[vertex_name][field_name]
```

Comment:
I just realized this style of dict is different from the others that receive a tuple of the arguments. I'll look into refactoring this.

Reply:
I also just noticed that the `self._histogram` variable is inconsistent with the init docstring.